easypost/__init__.py

```diff
@@ -321,27 +329,40 @@ def interpret_response(self, http_body, http_status):
             self.handle_api_error(http_status, http_body, response)
         return response
 
-    def requests_request(self, method, abs_url, headers, params):
+    def requests_request(self, method, abs_url, headers, params, isRating):
         method = method.lower()
         if method == 'get' or method == 'delete':
             if params:
                 abs_url = self.build_url(abs_url, params)
             data = None
         elif method == 'post' or method == 'put':
-            data = self.encode(params)
+            if not isRating:
+                data = self.encode(params)
+            else:
+                data = params
         else:
             raise Error("Bug discovered: invalid request method: %s. "
                         "Please report to contact@easypost.com." % method)
-
+        
         try:
-            result = requests_session.request(
-                method,
-                abs_url,
-                headers=headers,
-                data=data,
-                timeout=timeout,
-                verify=True,
-            )
+            if not isRating:
+                result = requests_session.request(
+                    method,
+                    abs_url,
+                    headers=headers,
+                    data=data,
+                    timeout=timeout,
+                    verify=True,
+                )
+            else:
+                result = requests_session.request(
+                    method,
+                    abs_url,
+                    headers=headers,
+                    data=json.dumps(data),
```

Comment:
this could also be done above on line 339 by just reassigning `data=json.dumps(data)`. it's simpler to keep all this edge-case logic in one place